android/app/src/main/kotlin/io/lantern/model/SessionModel.kt

```diff
@@ -185,7 +187,7 @@ class SessionModel(
                 val url = call.argument("url") ?: ""
                 url.isNotEmpty().let {
```

Comment:
The function block here will still be called even if url is empty. I think you want to do this instead..

```kotlin
if (url.isNotEmpty()) {
  val intent = Intent(activity, WebViewActivity_::class.java)
  // ...
}
```

Alternatively, to still use let..

```kotlin
url.takeIf{ it.isNotEmpty() }?.let {
  // ...
}
```